### PR TITLE
Parsing Python project metadata

### PR DESCRIPTION
**Pull request by NobleMathews**
### Python Support 

Adds support for python dep files

- Extracts relevant info
- Checks the following
   - requirements.txt,
   - setup.py
     - handles dynamic fields to an extent based on simple ast traversal
     - extends poetry parser with `with open()` logic handling to check for files in vcs step
   - setup.cfg
   - pyproject.toml
      - handles poetry based pyproject as well
- Tests added
- Extracted info to be changed once revised schema created

(Part 3 of pull requests to add support for parsing info from dependency files #10 )

### CODE REVIEW

File: dependencies/py/setup_reader.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+from configparser import ConfigParser
+from typing import Any
+from typing import Dict
+from typing import Iterable
+from typing import List
+from typing import Optional
+from typing import Union
+from poetry.core.semver import Version
+from poetry.core.semver import exceptions
+from poetry.utils.setup_reader import SetupReader
```
Comment (srijan-deepsource) [inline dependencies/py/setup_reader.py:16]:
```suggestion
from typing import (
    Any,
    Dict,
    Iterable,
    List,
    Optional,
    Union,
)
from poetry.core.semver import Version, exceptions
from poetry.utils.setup_reader import SetupReader
```

File: tests/test_dependency_files.py [hunk 2]
```diff
@@ -1,7 +1,43 @@
...
+        cfg_content = f.read()
+    result = dependencies.py.py_worker.handle_setup_cfg(cfg_content)
+    assert result["license"] == "MIT"
+
+
+def test_pyproject_toml():
+    with open("tests/data/example_pyproject.toml") as f:
+        pyproject = f.read()
+    result = dependencies.py.py_worker.handle_toml(pyproject)
+    assert result
```
Comment (srijan-deepsource) [inline tests/test_dependency_files.py:32]: Proper tests, please.

**Review by srijan-deepsource — COMMENTED** [review]
Need a couple of clarifications. Left comments.

Also, the tests need to be made better.

File: dependencies/py/setup_reader.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        install_requires = []
+        value = self._find_in_call(call, "install_requires")
+        if value is None:
+            # Trying to find in kwargs
+            kwargs = self._find_call_kwargs(call)
+
+            if kwargs is None or not isinstance(kwargs, ast.Name):
+                return install_requires
+
+            variable = self._find_variable_in_body(body, kwargs.id)
```
[the remark this answers is not in the document]
Comment (NobleMathews) [reply dependencies/py/setup_reader.py:74]: It's for when double splats are used
```
setupstuff = {
        "name":"PyGithub",
        "install_requires":[
            "deprecated",
            "pyjwt>=2.0",
            "pynacl>=1.4.0",
            "requests>=2.14.0",
        ]
    }
setuptools.setup(**setupstuff)
```

File: tests/test_dependency_files.py (hunk 2 above)
Comment (NobleMathews) [reply tests/test_dependency_files.py:32]: will update in latest pull as it has a proper definition of the schema there

File: tests/test_dependency_files.py [hunk 4]
```diff
@@ -1,7 +1,43 @@
...
+
+
+def test_poetry_toml():
```
[the remark this answers is not in the document]
Comment (NobleMathews) [reply tests/test_dependency_files.py:35]: missed it then but it's there in one of the next pulls 😅

**Review by srijan-deepsource — DISMISSED** [review]

**Review by srijan-deepsource — APPROVED** [review]

**Pull request merged**